src/main/java/com/tenx/ms/retail/order/service/OrderService.java

```diff
@@ -60,11 +60,11 @@ public Order createOrder(Long storeId, Order order) {
 
         List<OrderDetail> details = order.getProducts();
         if (details == null || details.isEmpty()) {
-            throw new ConstraintViolationException("Empty order details", null);
+            throw new NullArgumentException("Order products");
         }
 
         if (order.getClientId() == null) {
-            throw new ConstraintViolationException("Order must have a clientId", null);
+            throw new NullArgumentException("clientId");
```

Comment:
The Java standard library defines the [Objects.requireNonNull](http://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-) method.